File: FlightETS-Java/app/src/main/java/com/example/flightets/AuthApiService.java
package com.example.flightets;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.Headers;
import retrofit2.http.POST;

public interface AuthApiService {
    @POST("api/login")
    Call<LoginResponse> login(@Body LoginRequest request);

    @POST("api/search_flights")
    Call<List<Flight>> searchFlights(@Body SearchRequest request);
}

File: FlightETS/app/src/main/java/com/example/flightets/AuthApiService.kt
package com.example.flightets

import com.example.flightets.LoginRequest
import com.example.flightets.LoginResponse
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.Headers
import retrofit2.http.POST

interface AuthApiService {
    @Headers("Content-Type: application/json")
    @POST("login")
    fun login(@Body request: LoginRequest): Call<LoginResponse>

    @POST("api/search_flights")
    fun searchFlights(@Body request: SearchRequest): Call<List<Flight>>

}
